app/settings/tokens/edit/controller.ts

```diff
@@ -22,10 +22,10 @@ export default class SettingsTokensEditController extends Controller {
 
     @action
     refresh() {
-        this.clearTokenValue();
-
-        // Send action to route
-        this.send('refreshRoute');
+        if (this.token && this.token.tokenValue) {
+            this.token.tokenValue = undefined;
+            this.token.save();
```

Comment:
I believe the reason this works is because the API sends a response back when you do a PATCH request, and that response has the new information in it. It is not a proper way just to get a response, however. That's just relying on a side-effect of saving. What happens, for example, if this token is modified in another window, and you try to "refresh" the token this way? Does it overwrite the changes from the other window? 

Moreover, we don't want to delete the token value from the record stored on OSF (if that's allowed by the API). We only want our local view of the token value to be removed so that it doesn't show on screen.

Reply:
Ok, a security concern had arisen when the token was present. Current logic also relied on the token value, that when cleared rendered the other logical block on the template file. The changes were persisted to not allow changes to be rolled back and the token value viewable.

Reply:
But if the changes are persisted, then the user's token breaks forever, which is not what we want when we edit the scopes of a token. Changes should not be persisted, the token should just not be shown on-screen.

Reply:
Ok, and understood. It was commented that a security vulnerability was present when the changes were not persisted.